gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStore.js

```diff
@@ -0,0 +1,599 @@
+/*!
+GPII CouchDB OAuth 2 Data Store
+
+Copyright 2016 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+require("./DbDataStoreUtils.js");
+
+fluid.defaults("gpii.oauth2.dbDataSource", {
+    gradeNames: ["kettle.dataSource.URL", "kettle.dataSource.CouchDB"],
+    baseUrl: null,   // Supplied by integrators
+    termMap: {
+        dbName: null    // Supplied by integrators
+    },
+    notFoundIsEmpty: true,
+    rules: {
+        writePayload: {
+            "": ""
+        },
+        readPayload: {
+            "": ""
+        }
+    },
+    // Add baseUrl value into url template since baseUrl (an example baseUrl value: http://localhost:5984/)
+    // should not be passed into the "termMap" option because those termMap values will be encoded to
+    // compose the url.
+    url: {
+        expander: {
+            funcName: "fluid.stringTemplate",
+            args: ["%baseUrl:%port/%dbName%requestUrl", {
+                baseUrl: "{that}.options.baseUrl",
+                port: "{that}.options.port",
+                requestUrl: "{that}.options.requestUrl"
+            }]
+        }
+    }
+});
+
+fluid.defaults("gpii.oauth2.dbDataSource.writable", {
+    gradeNames: ["gpii.oauth2.dbDataSource", "kettle.dataSource.CouchDB.writable"],
+    writable: true,
+    writeMethod: "PUT"
+});
+
+fluid.defaults("gpii.oauth2.dbDataStore", {
+    gradeNames: ["gpii.oauth2.dataStore"],
+    // Supplied by GPII configuration to config all gpii.oauth2.dbDataSource instances.
+    // It contains these elements:
+    // 1. gradeNames: The mixin grade
+    // 2. baseUrl: The base URL to where the database is located. For example, a default locally installed CouchDB uses http://127.0.1.1
+    // 3. port: The port where the database is located. For example, a default locally installed CouchDB uses port 5984
+    // 4. termMap: {
+    //        dbName: The database name
+    //    }
+    dataSourceConfig: {
+    },
+    distributeOptions: [{
```

Comment:
Supply a name for this distribution using the more modern object rather than array syntax